int64 robotx
int64 roboty
int64 robotyaw